eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -261,53 +249,69 @@ where
         R: SampleRange<T> + Clone,
         C::Coeff: From<T>,
     {
-        // TODO: use Poly::coeffs_modify_include_zero() here and benchmark
         let mut res = Poly::non_canonical_zeroes(C::MAX_POLY_DEGREE);
-        for i in 0..C::MAX_POLY_DEGREE {
+        Poly::coeffs_modify_include_zero(&mut res, |coeff: &mut <C as PolyConf>::Coeff| {
             let coeff_rand = rng.gen_range(range.clone());
-            res[i] = coeff_rand.into();
-        }
+            *coeff = coeff_rand.into();
+        });
 
-        // Raw coefficient access must be followed by a truncation check.
-        res.truncate_to_canonical_form();
         res
     }
 
-    /// Plaintext addition is trivial
+    /// Sample a polynomial with random binnary coefficients, i.e. 0, 1
+    pub fn sample_binary_message(&self, rng: &mut ThreadRng) -> Message<C> {
+        let m = self.sample_uniform_range(0..=1_u64, rng);
+        Message { m }
+    }
+
+    /// Sample a polynomial with random ternary coefficients, i.e. -1, 0, 1, such that -1 is represented as C::T - 1
+    pub fn sample_ternary_message(&self, rng: &mut ThreadRng) -> Message<C> {
```

Comment:
Can this method be moved to [yashe/test.rs](https://github.com/Inversed-Tech/eyelid/tree/main/eyelid-match-ops/src/primitives/yashe/test.rs)?

Reply:
This can become a TODO comment.